### PR TITLE
Unit test - Apply feedback code review

### PR DESCRIPTION
**Pull request by ABTastyAdel**
Add unit tests

### CODE REVIEW

File: __tests__/FSStorage.test.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import {
+    getCacheFromPhone,
+    setModificationsCacheFromPhone
+} from '../lib/FSStorage';
+import FsLogger from '../lib/FsLogger';
+import AsyncStorage from '@react-native-community/async-storage';
+
+/// Data Mock , inspired from real case
+export const dataMock = {
```
Comment (Emidomenge) [inline __tests__/FSStorage.test.js:9]: put mock in a specific folder `mock` nearby unit test files

File: __tests__/FSStorage.test.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        apiKey: null,
+        initialModifications: null
+    }
+};
+
+export const emptyDataMock = {
+    fsModifications: []
+};
+
+it('checks getCacheFromPhone', async () => {
```
Comment (Emidomenge) [inline __tests__/FSStorage.test.js:47]: from line 47 until 94 wrap it into this:

```javascript
describe('Localstorage tests', () => {
     // Put the code here !
}
```

Comment (Emidomenge) [reply __tests__/FSStorage.test.js:47]: Use `describe` function allow us to categories tests correctly, avoiding to have everything flats.

Comment (Emidomenge) [inline __tests__/FSStorage.test.js:47]: `it('checks getCacheFromPhone', async () =`

into

`it('getCacheFromPhone - basic', async () =`

File: __tests__/FSStorage.test.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    return getCacheFromPhone(FsLogger.getLogger()).then((data) => {
+        expect(AsyncStorage.getItem).toBeCalledWith('@storage_fsModifications');
+
+        /// Check if is array
+        expect.arrayContaining(data);
+
+        // /// Read first object
+        const fsModif = data['modifications'][0]; /// Hard access , because we know the mock response don't have an empty array
+
+        console.log(fsModif);
```
Comment (Emidomenge) [inline __tests__/FSStorage.test.js:59]: remove console.log !

File: __tests__/FSStorage.test.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+        // /// VariationGroupId
```
Comment (Emidomenge) [inline __tests__/FSStorage.test.js:61]: Too many slashes in your JS commentary

`// /// VariationGroupId`

should be:

`// VariationGroupId`

Comment (Emidomenge) [reply __tests__/FSStorage.test.js:61]: Correct every line with this issue please

File: __tests__/FSStorage.test.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+        expect(values.RNBackGroundColor).toBe('pink');
+        expect(values.RN_btn_title).toBe('Save blog');
+        expect(values.RN_label_content).toBe('Content of post');
+        expect(values.RN_label_title).toBe('Title of post');
+        expect(values.btnTitle).toBe('Shop');
+    });
+});
+
+it('Test with Empty array', async () => {
```
Comment (Emidomenge) [inline __tests__/FSStorage.test.js:81]: `it('Test with Empty array', async () => {`

into

`it('should set correctly an empty array', async () => {`

File: __tests__/FSStorage.test.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    setModificationsCacheFromPhone(emptyDataMock, FsLogger.getLogger());
+    return getCacheFromPhone(FsLogger.getLogger()).then((data) => {
+        expect.arrayContaining(data['modifications']);
+        expect(data['modifications'].length).toBe(0);
+    });
+});
+
+it('Test with null', async () => {
```
Comment (Emidomenge) [inline __tests__/FSStorage.test.js:89]: `it('Test with null', async () => {`

into

`it('should set correctly null value', async () => {`

File: __tests__/flagshipHooks.test.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    useFsActivate,
+    useFsModifications
+} from '@flagship.io/react-sdk';
+
+import { renderHook, act } from '@testing-library/react-hooks';
+
+//// test hooks
+
+test('useFsModifications', () => {
+    const fsRequestModif = [
```
Comment (Emidomenge) [inline __tests__/flagshipHooks.test.js:13]: move it to the `mock`

File: coverage/clover.xml [hunk 8]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Emidomenge) [inline coverage/clover.xml:1]: maybe we can ignore `coverage` folder for the moment

File: __tests__/FSTools.test.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { checkValidityPatternForEnvId, maFct } from '../lib/FSTools';
+
+import { generateFlagshipId, maFctBis } from '../lib/FSTools';
+
+test('Testing with valide checkValidityPatternForEnvId', () => {
```
Comment (Emidomenge) [inline __tests__/FSTools.test.js:5]: if `test` is same as `it` then refacto and wrap with `describe`

File: __tests__/flagshipHooks.test.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+    expect(typeof result.current.stopBucketingPolling).toBe('function');
+
+    expect(typeof result.current.hiwt).toBeDefined();
+
+    expect(typeof result.current.status).toBeDefined();
+
+    const hitTest = result.current.hit;
+
+    const mockHit = {
```
Comment (Emidomenge) [inline __tests__/flagshipHooks.test.js:58]: move it

File: lib/FSStorage.js [hunk 11]
```diff
@@ -27,6 +27,7 @@ export const getCacheFromPhone = async (log) => {
 
     return new Promise((resolve) => {
         Promise.all([modificationsPromise, bucketPromise]).then((values) => {
+            console.log(values);
```
Comment (Emidomenge) [inline lib/FSStorage.js:30]: remove

File: coverage/clover.xml (hunk 8 above)
Comment (ABTastyAdel) [reply coverage/clover.xml:1]: Done

File: __tests__/FSStorage.test.js (hunk 1 above)
Comment (ABTastyAdel) [reply __tests__/FSStorage.test.js:9]: done

File: __tests__/FSStorage.test.js (hunk 2 above)
Comment (ABTastyAdel) [reply __tests__/FSStorage.test.js:47]: Done

File: __tests__/FSStorage.test.js (hunk 3 above)
Comment (ABTastyAdel) [reply __tests__/FSStorage.test.js:59]: Done

File: __tests__/FSStorage.test.js (hunk 4 above)
Comment (ABTastyAdel) [reply __tests__/FSStorage.test.js:61]: Done

File: __tests__/FSStorage.test.js (hunk 5 above)
Comment (ABTastyAdel) [reply __tests__/FSStorage.test.js:81]: done

File: __tests__/FSStorage.test.js (hunk 6 above)
Comment (ABTastyAdel) [reply __tests__/FSStorage.test.js:89]: done

**Pull request merged**